doc/src/challenges/0025-vecdeque.md

```diff
@@ -0,0 +1,77 @@
+# Challenge 25: Verify the safety of `VecDeque` functions
```

Comment:
This PR is missing an update to doc/src/SUMMARY.md to link to this new challenge.